gsa/src/web/pages/targets/component.js

```diff
@@ -15,406 +15,479 @@
+      excludeFileText = exclude_file.text().then(text => {
+        return text;
+      });
+    } else {
+      excludeFileText = Promise.resolve();
+    }
+
+    return Promise.all([fileText, excludeFileText]);
+  };
+
+  const handleSaveTarget = ({
+    alive_tests,
+    comment,
+    esxi_credential_id,
+    exclude_hosts,
+    exclude_file,
+    file,
+    hosts,
+    id,
+    in_use,
+    name,
+    port,
+    port_list_id,
+    reverse_lookup_only,
+    reverse_lookup_unify,
+    smb_credential_id,
+    snmp_credential_id,
+    ssh_credential_id,
+    target_exclude_source,
+    target_source,
+  }) => {
+    handleInteraction();
+
+    if (isDefined(id)) {
+      const {onSaved, onSaveError} = props;
+
+      return setFileText(file, exclude_file)
+        .then(text => {
+          const [fileText, excludeFileText] = text;
+
+          let mutationData;
+          if (in_use) {
+            mutationData = {
+              name,
+              comment,
+              aliveTest: alive_tests.toLowerCase(),
+              id,
+            };
+          } else {
+            mutationData = {
+              id,
+              aliveTest: alive_tests.toLowerCase(),
+              comment,
+              esxiCredentialId: esxi_credential_id,
+              hosts: target_source === 'file' ? fileText : hosts,
+              excludeHosts:
+                target_exclude_source === 'file'
+                  ? excludeFileText
+                  : exclude_hosts,
+              name,
+              sshCredentialPort: parseInt(port),
+              portListId: port_list_id,
+              reverseLookupOnly: reverse_lookup_only,
+              reverseLookupUnify: reverse_lookup_unify,
+              smbCredentialId: smb_credential_id,
+              snmpCredentialId: snmp_credential_id,
+              sshCredentialId: ssh_credential_id,
+            };
+          }
+
+          return modifyTarget(mutationData);
+        })
+        .then(onSaved, onSaveError)
+        .then(() => closeTargetDialog());
+    }
+
+    const {onCreated, onCreateError} = props;
+
+    return setFileText(file, exclude_file)
+      .then(text => {
+        const [fileText, excludeFileText] = text;
+
+        const mutationData = {
+          aliveTest: alive_tests.toLowerCase(),
+          comment,
+          esxiCredentialId: esxi_credential_id,
+          name,
+          hosts: target_source === 'file' ? fileText : hosts,
+          excludeHosts:
+            target_exclude_source === 'file' ? excludeFileText : exclude_hosts,
+          sshCredentialPort: parseInt(port),
+          portListId: port_list_id,
+          reverseLookupOnly: reverse_lookup_only,
+          reverseLookupUnify: reverse_lookup_unify,
+          smbCredentialId: smb_credential_id,
+          snmpCredentialId: snmp_credential_id,
+          sshCredentialId: ssh_credential_id,
+        };
+
+        return createTarget(mutationData);
+      })
+      .then(result => onCreated(result), onCreateError)
```

Comment:
can be simplified if the calling function/method passes to correct arguments already.
```suggestion
      .then(onCreated, onCreateError)
```